### PR TITLE
Add sad parameter #753

### PR DESCRIPTION
**Pull request by hendzeld**
Adds additional optional parameter BEAMSAD

### CODE REVIEW

**Comment by grzanka** [comment]
There was another PR just being merged: https://github.com/yaptide/ui/pull/885

Please merge master into your branch or rebase + force push on newest master

File: src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx [hunk 1]
```diff
@@ -131,6 +131,68 @@ function BeamSigmaField(props: { beam: Beam; onChange: (value: Beam['sigma']) =>
 	);
 }
 
+
+function BeamSadField(props: { beam: Beam; onChange: (value: Beam['sad']) => void }) {
+	const configuration = {
+		[SAD_TYPE.Rectangular]: {
+			X: {
+				text: 'sad in X'
```
Comment (grzanka) [inline src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx:139]:
```suggestion
				text: 'SAD X'
```

File: src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx [hunk 2]
```diff
@@ -131,6 +131,68 @@ function BeamSigmaField(props: { beam: Beam; onChange: (value: Beam['sigma']) =>
...
+			},
+			Y: {
+				text: 'sad in Y'
```
Comment (grzanka) [inline src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx:142]:
```suggestion
				text: 'SAD Y'
```

File: src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx [hunk 3]
```diff
@@ -131,6 +131,68 @@ function BeamSigmaField(props: { beam: Beam; onChange: (value: Beam['sigma']) =>
...
+		},
+
+	};
+
+	const selectedConfiguration = configuration[props.beam.sad.type];
+
+	return (
+		<>
+			<SelectPropertyField
+				label='Sad type'
```
Comment (grzanka) [inline src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx:161]:
```suggestion
				label='SAD planes'
```

File: src/ThreeEditor/util/Beam.ts [hunk 4]
```diff
@@ -13,7 +13,14 @@ export const SIGMA_TYPE = {
 	'Flat square': 'Flat square',
 	'Flat circular': 'Flat circular'
 } as const;
-export type SigmaType = keyof typeof SIGMA_TYPE;
+export type SigmaType = keyof typeof SIGMA_TYPE
+
+export const SAD_TYPE = {
+	'None' : 'None',
+	'Rectangular': 'Rectangular',
+	'Square': 'Square'
```
Comment (grzanka) [inline src/ThreeEditor/util/Beam.ts:21]: SAD dropdown menu should show:
'none',  'single (X=Y)',  'double (X and Y)'

**Review by grzanka — CHANGES_REQUESTED** [review]

File: src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx (hunk 1 above)
Comment (hendzeld) [reply src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx]: Should be OK now

File: src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx (hunk 2 above)
Comment (hendzeld) [reply src/ThreeEditor/components/Sidebar/propeteries/category/BeamConfiguration.tsx]: Should be OK now.
P.S.
Sorry, I uses plugin in VScode and it didn't show suggestions, so I changed it manually.

File: src/ThreeEditor/util/Beam.ts (hunk 4 above)
Comment (hendzeld) [reply src/ThreeEditor/util/Beam.ts]: Please check it is OK now.

**Review by grzanka — APPROVED** [review]

**Pull request merged**